docs/api_specification.md

```diff
@@ -281,4 +258,41 @@ Content-Type: application/json
 #### Response field description 
 same as above(docs/api_specification.md#post-v1apipayments)
 
----------------------------------------------------------------------------------------------------------------------------------------------------------
+## POST /v1/api/products/{productId}/payments
+
+This endpoint creates a new payment for a given product in pay-products.
+
+### Request example
+
+```
+POST /v1/api/products/uier837y735n837475y3847534/payments
+Authorization: Bearer API_TOKEN
+```
+
+### Response example
+
+```
+200 OK
+Content-Type: application/json
+{
+        "external_id": "h6347634cwb67wii7b6ciueroytw",
+        "next_url": "https://some.valid.url/paid",
+        "product_external_id": "uier837y735n837475y3847534",
+        "status": "CREATED",
+        "_links": [
+            {
+                "rel": "self",
+                "method": "GET",
+                "href": "https://govukpay-products.cloudapps.digital/v1/api/products/h6347634cwb67wii7b6ciueroytw"
```

Comment:
A few good points 👍 

- Updated the documentation to show a more representative example of `self` link. 
- Moved the `next_url` under the `_links` attributes for consistency across the project. 
- Added the `amount` captured during a payment. (This was not strictly necessary for the current feature but I can see it will be useful very soon so why not adding it while we are at it)
- You last point regarding `return_url` I would be inclined to not feature proof here since we don't have the answer yet.